source/client.ts

```diff
@@ -1100,7 +1192,15 @@ async function startInterpreter(
         shouldLog = false;
       }
 
-      let message: Message | null = null;
+      const lineMatch = line.match(COMPILING_PATTERN);
+      if (lineMatch) {
+        assert.ok(lineMatch[4]);
+        const uri = toAbsoluteUri(rootUri, lineMatch[4]);
+        collection.delete(uri);
```

Comment:
This is actually weird. When reloading, the "compiling" line comes before the diagnostic as expected. But when initially building, the "compiling" line comes _after_ the diagnostic. So the diagnostic gets cleared. 

This might be a consequence of the "compiling" messages coming from STDOUT while the diagnostics come from STDERR. At any rate, I'm not sure how to fix it.

Reply:
I think this may be related to `cabal repl --enable-multi-repl`. I can't reproduce this problem without that setup.